### PR TITLE
updated ReasonReact => 4.1, bs-platform to 3.0

### PR DESCRIPTION
**Pull request by idkjs**
**Pull Request Labels**

- [ ] feature
- [ ] blocking
- [x] docs

### CODE REVIEW

**Comment by Gregoirevda** [comment]
Thanks for this PR! I was waiting for feedback on the same sort of PR I did on bs-react-native, to be sure about backwards compatibility: https://github.com/reasonml-community/bs-react-native/pull/172#issuecomment-388785265

Didn't you had warnings with `Js.boolean` related changes? I think this should be updated too. `yarn clean && yarn start` to be sure to have latest warnings

**Comment by idkjs** [comment]
Will get on it.

**Comment by Gregoirevda** [comment]
Oh men! Thanks a lot!

**Comment by idkjs** [comment]
Not sure if this is something you wanted. Right now, `yarn start` on both src and example gets you a clean compile, no warnings or errors. I also updated some peer deps so definitely take a look at it or run tests if you have them.

File: package.json [hunk 1]
```diff
@@ -15,31 +15,26 @@
...
-    "apollo-cache-inmemory": "^1.1.5",
-    "apollo-client": "^2.2.0",
-    "apollo-link": "^1.0.7",
-    "apollo-link-http": "^1.3.2",
-    "bs-platform": "^2.1.0",
-    "graphql-tag": "^2.6.1",
-    "graphql_ppx": "^0.2.0",
-    "react-apollo": "^2.1.3",
-    "reason-react": "^0.3.1"
+    "apollo-cache-inmemory": "^1.2.1",
```
Comment (Gregoirevda) [inline package.json]: We shouldn't force the users of the library to have the latest version right?

**Comment by unicodeveloper** [comment]
@Gregoirevda any update on this?

File: package.json (hunk 1 above)
Comment (Gregoirevda) [reply package.json]: I had a bug yesterday with a previous version of apollo-client, so let's keep it like that ;)

**Comment by Gregoirevda** [comment]
@unicodeveloper I need to test it and will then merge and deploy

**Pull request closed without merging**